base/complex.jl

```diff
@@ -1141,3 +1141,16 @@ function complex(A::AbstractArray{T}) where T
     end
     convert(AbstractArray{typeof(complex(zero(T)))}, A)
 end
+
+## Promotion to complex ##
+
+_default_type(T::Type{Complex}) = Complex{Int}
+
+## Machine epsilon for complex ##
+
+function eps(z::Complex{T}) where {T<:AbstractFloat}
+    x, y = eps(real(z)), eps(imag(z))
+    hypot(x, y)
+end
```

Comment:
Can be shortened to:
```suggestion
eps(z::Complex) = hypot(eps(real(z)), eps(imag(z)))
```